tests/utils.py

```diff
@@ -0,0 +1,49 @@
+import requests
+import hashlib
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+from pathlib import Path
+
+files_directory = Path(__file__).parent / 'files'
+
+def setup_reference_fmus():
+    """
+        This function downloads reference FMUs from the Modelica group and unpacks
+        them into the test files directory.
+        The MD5sum of the URL is checked in order to avoid unnecessary downloading.
+        Note that this requires an internet connection to work.
+    """
+
+    def download_url(url, save_file_to, chunk_size=1024):
+        """ Download file from URL to 'save_file_to' in chunks. """
+        r = requests.get(url, stream=True)
```

Comment:
What happens if this fails, e.g., files were removed or no internet connection? 

I suppose having an assert on the return status (if there is one) would be OK here.

Reply:
I have replaced the use of `requests` with `urllib` instead (part of Python standard library).
There is now also some exception handling that I have tested by turning off my internet connection.